src/core/egglog-herbie.rkt

```diff
@@ -44,6 +44,10 @@
 
   (define egglog-file-path
     (let ([temp-file (make-temporary-file "program-to-egglog-~a.egg")])
+      ; temp-file is properly garbage collected if it exists
+      (when (file-exists? temp-file)
+        (delete-file temp-file))
+
       (with-output-to-file temp-file #:exists 'replace (lambda () (for-each writeln curr-program)))
```

Comment:
Is it not the case that `make-temporary-file` guarantees uniqueness? I would double check the docs. It seems weird to delete the file given that `with-output-to-file` with `#:exists 'replace` makes sure to overwrite any existing file. Something is a bit fishy about this code, but not a big deal to leave the code in.